File: HasCubeRoot.java
/**
 * Performance (time)
 * real	0m0.337s
 * user	0m0.299s
 * sys	0m0.045s
 */
public class HasCubeRoot {

	public static boolean hasCubeRoot(int x) {
		for (int i = 0, sum = 0; sum <= x; i++, sum = i*i*i) {
			if (sum == x) return true;
		}
		return false;
	}


	public static void main(String... args) {
		for (int i = 0; i < 1000000; i++) {
			if (hasCubeRoot(i)) {
				System.out.println(i + " has a cube root");
			} 
		}	
	}
}

File: has-cube-root.rb
# Performance (time)
# real	0m17.562s
# user	0m17.007s
# sys	0m0.041s

def has_cube_root? x
	raise new ArgumentError if x < 0
	y = 0
	while y**3 <= x
		return [true, y] if y**3 == x
		y += 1
	end 
	return false
end

0.upto(1000000) do |x|
	answer = has_cube_root? x
	puts "#{x} has a cube root %s" % answer[1] if answer
end

